core/server/services/auth/session/index.js

```diff
@@ -0,0 +1,130 @@
+const {URL} = require('url'),
+    common = require('../../../lib/common'),
+    config = require('../../../config'),
+    settingsCache = require('../../settings/cache'),
+    {User} = require('../../../models/user'),
+    {Session} = require('../../../models/session'),
+    session = require('express-session'),
+    SessionStore = require('./store'),
+    {BadRequestError, UnauthorizedError, InternalServerError} = require('../../../lib/common/errors');
+
+const getOrigin = function getOrigin(req) {
+    const origin = req.get('origin');
+    const referrer = req.get('referrer');
+
+    if (!origin && !referrer) {
+        return null;
+    }
+
+    if (origin) {
+        return origin;
+    }
+
+    try {
+        return new URL(req.get('referrer')).origin;
+    } catch (e) {
+        return null;
+    }
+};
+
+const createSession = function createSession(req, res, next) {
```

Comment:
This feels like a controller 🤔

Reply:
100% agree - because we're treating sessions as a resource. Though because that method is essentially the `login` method I still think it sits in the auth service. The oauth logins are handled in the auth service too. not 100% convinced either way tbh :thought_balloon:

Reply:
> The oauth logins are handled in the auth service too

Yeah but this logic is really old 😁It was only refactored once since i started and we grouped every auth related operation into a service.

> 100% agree - because we're treating sessions as a resource

👍 

The tricky thing here is that API v2 controllers are not a thing yet. 

I might be able to prepare an empty folder structure for v2 today, which just makes it possible to add single additions.

e.g. `/api/v2/sessions`

Reply:
The only thing is that this controller won't be using the models directly, it'll be setting `req.session` and calling `req.session.destroy` are you good with that? If so, i'll get `api/v1/sessions` in A$AP

Reply:
Wouldn't the controller use

> User.check({

> api/v1/sessions in A$AP

You probably mean /v2/?

I'll prepare an empty v2 controller structure now.

Reply:
lal yup - okay thanks

Reply:
> Wouldn't the controller use

>>    User.check({

Yes, but it would be Session controller right? But wouldn't directly use the `Session` models.

Reply:
Yeah, hmmmmmmmmmmmmmmm, i would still put it to controllers? 
e.g. sending the status code from the auth service is so weird (but i agree with you, the old impl did the same)

Reply:
We agreed. We keep it as is for now. There is no nice way to solve this with a controller, because of `req.session`. We can only solve it nicely if we remove `express-session`.

Reply:
:dancer: